runtime/jcl/common/mgmtmemory.c

```diff
@@ -702,9 +703,72 @@ Java_com_ibm_lang_management_internal_MemoryNotificationThread_processNotificati
 				return;
 			}
 
-			(*env)->CallVoidMethod(env, threadInstance, helperGCID,
-					gcName, gcAction, gcCause,
-					(jlong)gcInfo->index, (jlong)gcInfo->startTime, (jlong)gcInfo->endTime,
+			/* Error tracepoint for the cases
+			 * the value of init or max is negative but not -1; or
+			 * used is greater than the value of committed; or
+			 * committed is greater than the value of max if max is not -1.
+			 */
+			if (TrcEnabled_Trc_JCL_com_ibm_lang_management_internal_MemoryNotificationThread_processNotificationLoop_memoryUsageError
+			|| TrcEnabled_Trc_JCL_com_ibm_lang_management_internal_MemoryNotificationThread_processNotificationLoop_memoryUsage
+			) {
+				const char *nsGcName = (*env)->GetStringUTFChars(env, gcName, NULL);
+				for (idx = 0; idx < gcInfo->arraySize; ++idx) {
+					U_64 preUsed = gcInfo->preUsed[idx];
+					U_64 preCommitted = gcInfo->preCommitted[idx];
+					I_64 preMax = gcInfo->preMax[idx];
+					U_64 postUsed = gcInfo->postUsed[idx];
+					U_64 postCommitted = gcInfo->postCommitted[idx];
```

Comment:
All of these should probably be `I_64` instead of `U_64` as they correspond to fields in `MemoryUsage` of type `long`; but that should be addressed separately.